server/lobbyconnection.py

```diff
@@ -229,7 +229,8 @@ async def command_matchmaker_info(self, message):
     async def send_game_list(self):
         await self.send({
             "command": "game_info",
-            "games": [game.to_dict() for game in self.game_service.open_games]
+            "games": [game.to_dict() if game.is_visible_to_player(self.player) else None
+                      for game in self.game_service.open_games]
```

Comment:
The filter condition should come after the `for x in y` part, then you don't need to include `None` objects:

```python3
>>> my_list = [1, 123, 5, 20, 3, 10, 11, 7]

>>> [item for item in my_list if item < 10]
[1, 5, 3, 7]

>>> [item if item < 10 else None for item in my_list]
[1, None, 5, None, 3, None, None, 7]
```

Reply:
Alright, that makes sense, though the python syntax is a bit confusing here.